src/btree/bt_read.c

```diff
@@ -331,7 +331,8 @@ __evict_force_check(WT_SESSION_IMPL *session, WT_REF *ref)
 	 * If we have already tried and the transaction state has not moved on,
 	 * eviction is highly likely to fail.
 	 */
-	if (page->modify->last_eviction_id == __wt_txn_oldest_id(session))
+	if (page->memory_footprint < btree->maxmempage &&
+	    page->modify->last_eviction_id == __wt_txn_oldest_id(session))
```

Comment:
Can you please add a sentence to the comment above?  Something like "if the page grows above the maximum size, keep trying to split it or inserts of large records can lead to extremely large memory footprints".